ephios/core/signup/methods.py

```diff
@@ -126,6 +127,37 @@ def reverse_event_detail(self, event):
         return event.get_absolute_url()
 
 
+@dataclasses.dataclass(frozen=True)
+class ForeignParticipant(AbstractParticipant):
+    def new_participation(self, shift):
+        return ForeignParticipation(
+            shift=shift, first_name=self.first_name, last_name=self.last_name
+        )
+
+    def participation_for(self, shift):
+        try:
+            return ForeignParticipation.objects.get(
+                shift=shift, first_name=self.first_name, last_name=self.last_name
+            )
+        except ForeignParticipation.DoesNotExist:
+            return None
+
+    def all_participations(self):
+        return AbstractParticipation.objects.none()
+
+    def reverse_signup_action(self, shift):
+        return reverse("core:signup_action", kwargs=dict(pk=shift.pk))
+
+    def reverse_event_detail(self, event):
+        return event.get_absolute_url()
```

Comment:
Do these need to be implemented? And if, shouldn't it return something that makes it obvious that it's not a meaningful thing to do?